components/user-details.tsx

```diff
@@ -0,0 +1,143 @@
+"use client";
+
+import { useEffect, useState } from "react";
+import ModeEdit from "@mui/icons-material/ModeEdit";
+import Check from "@mui/icons-material/Check";
+import { useSession } from "next-auth/react";
+import { Tooltip } from "@mui/material";
+
+interface User {
+  name: string;
+  email: string;
+  role: string;
+  createdAt: Date;
+}
+
+export const UserDetails = () => {
```

Comment:
Let's do one more iteration towards NextJs application :) 
There are a way too many states for "almost static" page. For example, list of users can be pre-fetched on server side. useEffect is not needed at all. Session should be handled on the backend as currently there is no API protection which mean that I can update user without admin rights.
updateUser can be server action where you can call db directly.
Same applies to user list.
Also, it is preferred to have some split between higher order components (HOC) and UI components (dumb components). Please have a look into this pattern and update code accordingly.